darwin/future/core/types/query.py

```diff
@@ -13,40 +27,64 @@ class QueryFilter(DefaultDarwin):
     ----------
     name: str
     param: str
+    modifier: Optional[Modifiers]: Optional modifier to apply to the filter
     """
 
     name: str
     param: str
+    modifier: Optional[Modifiers] = None
+
+    def filter_attr(self, attr: Any) -> bool:  # type: ignore
+        caster: Callable[[str], Any] = type(attr)  # type: ignore
+        param = caster(self.param)  # attempt to cast the param to the type of the attribute
+        if self.modifier is None:
+            return attr == param
+        elif self.modifier == Modifiers.GREATER_EQUAL:
+            return attr >= param
+        elif self.modifier == Modifiers.GREATER_THAN:
+            return attr > param
+        elif self.modifier == Modifiers.LESS_EQUAL:
+            return attr <= param
+        elif self.modifier == Modifiers.LESS_THAN:
+            return attr < param
+        elif self.modifier == Modifiers.NOT_EQUAL:
+            return attr != param
+        elif self.modifier == Modifiers.CONTAINS:
+            return param in attr
+        else:
+            raise ValueError(f"Unknown modifier {self.modifier}")
 
 
-class Query:
+class Query(Generic[T], ABC):
```

Comment:
Gotta love them Generic TypeVars, never cause any trouble with mypy. 😉

Reply:
getting this typing to work with mpyp was a major PITA actually and huge props to chatgpt for figuring out where I was going wrong.